### ISSUE TITLE
`unawaited_futures` false negative

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/linter#4260]
**Describe the issue**
`unawaited_futures` not linting with `FutureOr`

**To Reproduce**
Add the lint to `analysis_options.yaml` and run the following:

```dart
void main() async {
  futureOrFunc(); //Nothing
  futureFunc(); //Linted
}

Future<void> futureFunc() async {}

FutureOr<void> futureOrFunc() async {}
```

**Expected behavior**
I expected the linter to warn me about `FutureOr Function`s so that I only process things after the previous lines have been processed.

**Comment by lrhn** [comment]
The current [description](https://dart-lang.github.io/linter/lints/unawaited_futures.html) of `unawaited_futures` only say that it warns about things with type `Future`, not `FutureOr`.

So it's doing what it claims it's doing.
It's not unreasonable to await `FutureOr` too, but be aware that [`unawaited`](https://api.dart.dev/stable/2.14.1/dart-async/unawaited.html) only accepts a `Future` as argument, so you can't do `unawaited(futureOrFunc());` to disable a lint applying to `FutureOr`s.
